test/matrix_generator.cc

```diff
@@ -924,65 +926,37 @@ void decode_matrix(
         throw std::exception();
     }
 
-    if (type == TestMatrixType::zero      ||
-        type == TestMatrixType::one       ||
-        type == TestMatrixType::identity  ||
-        type == TestMatrixType::jordan    ||
-        type == TestMatrixType::circul    ||
-        type == TestMatrixType::fiedler   ||
-        type == TestMatrixType::gfpp      ||
-        type == TestMatrixType::orthog    ||
-        type == TestMatrixType::riemann   ||
-        type == TestMatrixType::ris       ||
-        type == TestMatrixType::randb     ||
-        type == TestMatrixType::randn     ||
-        type == TestMatrixType::rands     ||
-        type == TestMatrixType::rand)
-    {
-        // warn first time if user set cond and matrix doesn't use it
-        static std::string last;
-        if (! cond_default && last != kind) {
-            last = kind;
-            fprintf( stderr, "%sWarning: matrix '%s' ignores cond %.2e.%s\n",
-                     ansi_red, kind.c_str(), params.cond(), ansi_normal );
-        }
-        params.cond_used() = testsweeper::no_data_flag;
+    // Check if cond is known or unused.
+    if (type == TestMatrixType::zero || type == TestMatrixType::one) {
+        cond = inf;
     }
-    else if (dist == TestMatrixDist::randn ||
-             dist == TestMatrixDist::rands ||
-             dist == TestMatrixDist::rand)
-    {
-        // warn first time if user set cond and distribution doesn't use it
-        static std::string last;
-        if (! cond_default && last != kind) {
-            last = kind;
-            fprintf( stderr, "%sWarning: matrix '%s': rand, randn, and rands "
-                     "singular/eigenvalue distributions ignore cond %.2e.%s\n",
-                     ansi_red, kind.c_str(), params.cond(), ansi_normal );
-        }
-        params.cond_used() = testsweeper::no_data_flag;
+    else if (type == TestMatrixType::identity) {
```

Comment:
We could also add `orthog` here since it's also orthogonal.

Reply:
Yes, done.